api/src/main/java/uk/gov/hmcts/payment/api/scheduler/PaymentsReportService.java

```diff
@@ -204,9 +205,21 @@ private void generateCsvAndSendEmail(List<PaymentCsvDto> payments, String paymen
         for (PaymentCsvDto payment : payments) {
             List<FeeCsvDto> fees = payment.getFees();
             for (FeeCsvDto unPopulatedFee : fees) {
-                Fee2Dto fee = feesDataMap.get(unPopulatedFee.getCode());
-                unPopulatedFee.setMemoLine(fee.getMemoLine());
-                unPopulatedFee.setNaturalAccountCode(fee.getNaturalAccountCode());
+                Fee2Dto feeFromFeesRegister = feesDataMap.get(unPopulatedFee.getCode());
+                if(null!=feeFromFeesRegister) {
+                    if(unPopulatedFee.getVersion().equals(feeFromFeesRegister.getCurrentVersion().getVersion())) {
```

Comment:
pls remove this whole method. in toCSV it should retrieve individual value from map i.e. lazy initialization from fees register at repo level only when needed